modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BJSONType.java

```diff
@@ -28,6 +29,8 @@
  */
 public class BJSONType extends BType {
 
+    protected StructDef constraint;
```

Comment:
Just realised, having the schema on varRef only won't be enough. There can be other Expressions that can return a json with schema.